### PR TITLE
Gh 222 remove closeable

### LINKED ISSUE #222: Remove Closable
Action the deprecation of closable from gh-220.
Closable is used with scaling big data and doesn't belong in Koryphe.
The closable classes conflict with class exception with the counterparts in Gaffer.

This is breaking change for classes that will error in gaffer anyway and likly can't be used.

### PR DESCRIPTION
**Pull request by r32575**
This PR:
- removes `CloseableIterable`, `CloseableIterator` and all `iterable` classes
- changes all `CloseableIterable` and `CloseableIterator` references to `Iterable`/`Iterator`
- fixes tests

# Related Issue

- Resolve #222

### CODE REVIEW

File: core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java [hunk 1]
```diff
@@ -91,38 +94,32 @@ public void shouldErrorIfObjectsAreTheWrongType() {
...
 
         // Then
-        try {
-            boc.apply(stateTuple, inputTuple);
-            fail("Expected aggregation to fail");
-        } catch (ClassCastException e) {
-            assertNotNull(e.getMessage());
-        }
-
+        final ClassCastException actual = assertThrows(ClassCastException.class,
```
Comment (GCHQDev404) [inline core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java:97]: 👍

**Review by GCHQDev404 — CHANGES_REQUESTED** [review]

File: core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java [hunk 2]
```diff
@@ -15,29 +15,33 @@
...
     @Override
+    @SuppressWarnings("unchecked")
     public void shouldJsonSerialiseAndDeserialise() throws IOException {
         // Given
-        TupleAdaptedBinaryOperatorComposite instance = getInstance();
-        String json = "" +
+        final TupleAdaptedBinaryOperatorComposite<Object> instance = getInstance();
+        final String json = "" +
+                "{" +
+                "\"operators\": [" +
```
[the remark this answers is not in the document]
Comment (r32575) [reply core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java:31]: Yeah, the IDE altered them when I saved... I'll redo

[the remark this answers is not in the document]
Comment (r32575) [reply core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java]: Reformatted the code

File: core/src/main/java/uk/gov/gchq/koryphe/impl/function/CsvToMaps.java [hunk 3]
```diff
@@ -62,12 +61,14 @@
             return null;
         }
 
-        try {
-            final CSVParser csvParser = new CSVParser(new StringReader(csv), getCsvFormat());
-            final CloseableIterable<CSVRecord> csvRecords = IterableUtil.limit(csvParser.getRecords(), firstRow, null, false);
+        Iterable<CSVRecord> csvRecords = null;
+        try (final CSVParser csvParser = new CSVParser(new StringReader(csv), getCsvFormat())) {
```
[the remark this answers is not in the document]
Comment (r32575) [reply core/src/main/java/uk/gov/gchq/koryphe/impl/function/CsvToMaps.java:65]: Not sure that Gaffer is ready to upgrade to Java 11 yet but the change is required it. May be better to create a `Koryphe v2` with the change in for Java 8 and make the change in the #239 or create another issue.

File: core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java [hunk 4]
```diff
@@ -15,29 +15,33 @@
...
+        final TupleAdaptedBinaryOperatorComposite<Object> instance = getInstance();
+        final String json = "" +
                 "{" +
                     "\"operators\": [" +
                         "{" +
                             "\"selection\": [\"input\", \"anotherInput\"]," +
                             "\"binaryOperator\": {" +
                                 "\"class\": \"uk.gov.gchq.koryphe.impl.binaryoperator.Sum\"" +
-                            "}" +
+                             "}" +
```
Comment (GCHQDeveloper314) [inline core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java:36]: What's this? It looks to me like the indentation was correct before this change.

**Review by GCHQDev404 — APPROVED** [review]
No Objections

**Review by t11947 — APPROVED** [review]

**Comment by t11947** [comment]
We'll stick with the Java 8 syntax for now then.

File: core/src/main/java/uk/gov/gchq/koryphe/util/IterableUtil.java [hunk 5]
```diff
@@ -270,10 +261,8 @@ public void close() {
 
     /**
      * @param <T> the type of items in the iterator
-     * @deprecated Closable will be removed, it is used with scaling Big Data and does not belong in Koryphe.
      */
-    @Deprecated
```
[the remark this answers is not in the document]
Comment (t92549) [reply core/src/main/java/uk/gov/gchq/koryphe/util/IterableUtil.java:275]: This is indeed being un-Deprecated and as this PR now fixes or deletes all previously Deprecated code, #237 has been closed

**Review by t92549 — APPROVED** [review]

File: core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (t92549) [reply core/src/test/java/uk/gov/gchq/koryphe/tuple/binaryoperator/TupleAdaptedBinaryOperatorCompositeTest.java]: For me it still shows unindented

**Pull request merged**